src/js/jsx/sections/layers/LayersPanel.jsx

```diff
@@ -467,7 +467,7 @@ define(function (require, exports, module) {
 
                 var dragSource = collection.pluck(this.state.dragTargets, "id");
 
-                flux.actions.layers.reorder(doc, dragSource, dropIndex)
+                return flux.actions.layers.reorder(doc, dragSource, dropIndex)
```

Comment:
the `then` part is from the master, but I agree with you on that. Let me see if I can move to the action safely.